### DISCUSSION TITLE
Will Hasura's push to Hasura DDN v3 impact Nhost in anyway?

### DISCUSSION THREAD

**Discussion by ammarfaris** [nhost/nhost#2841] [General]
Hi there, a quick question, with Hasura DDN (v3) in GA just recently (https://hasura.io/blog/launching-hasura-ddn), will it impact Nhost in anyway? 

Also, is Hasura v3 sort of "backward-compatible" with Hasura v2, where hasura-auth and hasura-storage will continue to work without any modifications? or will there be major changes needed from Nhost's side?

Just trying to understand the situation with Hasura v2 (how long will it continue to be supported by Hasura the company, compatibility, etc) vs. Hasura DDN (v3), as Nhost is heavily linked to Hasura...

Thanks in advance!

**Comment by dbarrosop** [comment]
> will it impact Nhost in anyway?

No impact whatsoever, what works today will work in the future.

> is Hasura v3 sort of "backward-compatible" with Hasura v2?

No, Hasura DDN is a complete different product.

> how long will it continue to be supported by Hasura the company

I think they were saying 1-2 more years of support but our plan is to fork it and keep supporting it ourselves when that happens.

> or will there be major changes needed from Nhost's side?

Given that Hasura DDN isn't opensource (unless I am mistaken some components are but the core of it isn't) we don't have plans to support it.

**Comment by ammarfaris** [comment]
Thank you for your swift response @dbarrosop .

> I think they were saying 1-2 more years of support but our plan is to fork it and keep supporting it ourselves when that happens.

Glad to know that at worst-case-scenario, you guys are willing to fork v2 👍🏻

I looked at their v3 FAQ (https://hasura.io/docs/3.0/support/faq/):

_Will Hasura v2 be supported after Hasura DDN is released?
We will still support, maintain and improve Hasura v2 as per our [version support policy](https://hasura.io/docs/latest/policies/versioning/) and will be providing migration tools to move Hasura v2 deployments to DDN in the near future._

_Can we request to lock ourselves to 2.x?
The migration is an opt-in process, so you do not need to request to have yourselves locked to a particular version. If however, you see no upgrade path even in the next two years, please give us a heads-up and we will work out options with you._

and also looked at[ a discussion on their official Discord](https://discord.com/channels/407792526867693568/1232075847901646928) for the topic "When will Hasura v2 enter End-of-Life?"?

_Question: I just did a google search for Hasura EOL and was directed to this page. The latest version listed is 2.36 with an EOL for Dec 2025. Is the end of 2025 the EOL for newer 2.x versions as well? https://hasura.io/docs/2.0/policies/versioning/_

_Answer: Regarding EOL, The 2.36 is the latest published LTS. We will be either publishing a new LTS or extending the lifetime for this LTS for sure. 2.x versions will still happen (Security fixes, Community contributions, etc); but obviously brand new or architecturally complex features would be prioritised in V3._

> Given that Hasura DDN isn't opensource (unless I am mistaken some components are but the core of it isn't) we don't have plans to support it.

It says in the v3 FAQ that the engine and data connectors are open-sourced, but i'm not sure how much work is needed for nhost to make the existing components compatible with v3 components, and if some v3 components are behind a paywall (they ask to contact sales for Self-Hosted v3 option - https://hasura.io/docs/3.0/deployment/private/self-hosted); 

[v3 FAQ]
_Is Hasura DDN open source?
The [Hasura DDN engine](https://github.com/hasura/graphql-engine/tree/master/v3) and all [Hasura data connectors](https://hasura.io/connectors) are open-source and can be run on your own infrastructure. With the optimizations and features that Hasura DDN offers at the price point, our cloud service is the best way to use Hasura and self-hosting should be an option when specific requirements are needed and enterprise policies dictate._
